esahg/esahg.php

```diff
@@ -1,149 +1,170 @@
-			}
-			global $wp_filesystem;
-			$unzipfile = unzip_file( $download_to . '/' . $zip_filename, $download_to );
-			if ( is_wp_error( $unzipfile ) ) {
-				echo $unzipfile->get_error_message();
-			} else {
-				echo '<p>' . __( 'WordPress zip file unpacked.', 'esahg' ) . '</p>';
-				$files = array();
-				$files = walk_wp_directory( $download_to . '/wordpress' );
-				$hashes = hash_files( $files );
-			}
-			$wp_filesystem->delete( $download_to . '/wordpress/', true );
-		} else {
-			echo '<p>' . __( 'Sorry, unable to locate the WordPress zip archive.', 'esahg' ) . '</p>';;
-		}
-	}
-
-	if ( ! $wp_filesystem->put_contents( $es_path . $hash_filename, $hashes, FS_CHMOD_FILE ) ) {
-		echo '<p>' . __( 'Unable to write hashes file please create it manually with the following content:', 'esahg' ) . '</p>';
-		echo '<textarea>' . $hashes . '</textarea>';
-	} else {
-		echo '<p>' . sprintf( __( 'A Hash file called %s was successfully created!', 'esahg' ), $hash_filename ) . '</p>';
-	}
-	echo '</div>';
-} // end eashg_menu()
-
-function walk_wp_directory( $dir ) {
-	global $download_to, $files;
-	if ( $handle = @opendir( $dir ) ) {
-		while ( false !== ( $file = readdir( $handle ) ) ) {
-			if ( '.' !== $file && '..' !== $file ) {
-				$file = $dir . '/' . $file;
-				if ( is_dir( $file ) ) {
-					walk_wp_directory( $file );
-				} elseif ( is_file( $file ) ) {
-					if ( false === strstr( $file, 'wp-content/plugins/' ) && false === strstr( $file, 'wp-content/themes/' ) ) {
-						$files[] = $file;
-					}
-				}
-			}
-		}
-		closedir( $handle );
-		return $files;
-	} else {
-		echo '<p>' . sprintf( __( 'Apologies, it seems the zip file has disappeared, please manually download it to %s or try again.', 'esahg' ), ESAHGPATH );
-	}
-} // end walk_wp_directory()
-
-function hash_files( $files = array() ) {
-	if ( empty( $files ) ) { return false; }
-
-	global $download_to;
-	sort( $files );
-	$file_hashes = '<?php' . "\r\n" . '$filehashes = array(' . "\r\n";
-	foreach ( $files as $file ) {
-		$file_hashes .= "'" . str_replace( $download_to . '/wordpress/', '', $file ) . "' => '" . md5( $file ) . "',\r\n";
-	}
-	$file_hashes .= ");\r\n?>";
-
-	return $file_hashes;
+<?php
+/*
+Plugin Name: Exploit Scanner Auto Hash Generator
+Plugin URI: https://github.com/mattyrob
+Description: Checks for and then automatically generates hash files for use in the Exploit Scanner plugin for WordPress.
+Version: 1.0b
+Author: Matthew Robinson
+Author URI: https://github.com/mattyrob
+Licence: GPL3
+Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=2387904
+Text Domain: esahg
+*/
+
+define( 'ESAHGDIR', trailingslashit( plugin_basename( dirname( __FILE__ ) ) ) );
+define( 'ESAHGPATH', WP_CONTENT_DIR . '/' . ESAHGDIR );
+add_action( 'admin_menu', 'esahg_admin' );
+
+function esahg_admin() {
+	load_plugin_textdomain( 'eashg', 'wp-content/plugins/' . ESAHGDIR, '/' . ESAHGDIR );

```

Comment:
That function only exists inside the WP File System and it's not worth instantiating that so early just to use that function.